src/core/db/repository/base.py

```diff
@@ -65,13 +65,30 @@ async def create_all(self, objects: list[DatabaseModel]) -> None:
         self._session.add_all(objects)
 
 
-class ContentRepository(AbstractRepository):
+class ContentRepository(AbstractRepository, abc.ABC):
     @auto_commit
-    async def archive_all(self) -> None:
-        """Добавляет все объекты модели в архив."""
-        await self._session.execute(update(self._model).values({"archive": True}))
-
-    async def get_all_ids(self) -> list[int]:
-        """Возвращает id всех объектов модели из базы данных."""
-        ids = await self._session.execute(select(self._model.id))
-        return ids.scalars().all()
+    async def set_update_is_archived_false_to_true(self, ids: list[int]) -> None:
+        """Изменяет is_archived с False на True у не указанных ids."""
+        await self._session.execute(
+            update(self._model)
+            .where(
+                and_(
+                    self._model.is_archived == False,
+                    not_(self._model.id.in_(ids))
+                )
+            )

```

Comment:
Как по мне, двойной where более читабельным чем `and_`. Надо будет так на всем проекте поправить, но пока только у тебя.
```suggestion
            .where(self._model.is_archived == False)
            .where(self._model.id.not_in(ids))
```

Reply:
Понял. Поменяю.

Reply:
Только вроде not_in вроде нет команды, у меня ошибку выдавал. А к notin не было вопросов)